internal/controller/pipelinerollout/pipelinerollout_ppnd.go

```diff
@@ -201,6 +202,29 @@ func (r *PipelineRolloutReconciler) needPPND(ctx context.Context, pipelineRollou
 	return &needPPND, nil
 }
 
+// Determine if the Pipeline has changed and needs updating
+// We need to ignore any field that could be set by Numaplane in the pause-and-drain process
+// TODO: do we really need this or can we just use usde functionality and just look for data loss fields?
+func (r *PipelineRolloutReconciler) pipelineNeedsUpdatingForPPND(ctx context.Context, from, to *unstructured.Unstructured) (bool, error) {
+	numaLogger := logger.FromContext(ctx)
+	fromCopy := from.DeepCopy()
+	toCopy := to.DeepCopy()
+	// remove lifecycle.desiredPhase field from comparison to test for equality
+	numaflowtypes.PipelineWithoutDesiredPhase(fromCopy)
+	numaflowtypes.PipelineWithoutDesiredPhase(toCopy)
```

Comment:
For this function, we ignore `lifecycle.desiredPhase`. For the other function we ignore fields changed when doing Progressive upgrade (e.g. `scale`)